[AG-11746] Fix(ssrm): allow calling api resetRowHeights

sdwvit opened the pull request:
Fixes [AG-11746](https://ag-grid.atlassian.net/browse/AG-11746)

<img width="919" height="982" alt="Screenshot 2025-08-28 at 12 40 40" src="https://github.com/user-attachments/assets/efe4e373-1532-478d-8c50-96a499d99ef0" />

sdwvit committed:
Add resetRowHeights method to row models and update module exports

 - Add _getViewportRowModel to retrieve viewport row model instance
 - Introduce ViewportRowModelApiModule for API functions in viewport row model
 - Implement resetRowHeights in ViewportRowModel to reset row heights
 - Update module name exports to include ViewportRowModelApi
 - Add resetRowHeights API function in server-side row model to trigger viewport row height reset
 - Update enterprise module name validation to include ViewportRowModelApi
 - Define _ViewportRowModelGridApi interface with resetRowHeights method
 - Register resetRowHeights in API function validation for client, server, and viewport row models

sdwvit committed:
Implement resetRowHeights in server-side and viewport row models

 - Add resetRowHeights to server-side row model with beans parameter
 - Implement resetRowHeights in viewport row model with console log
 - Remove server-side API functions from viewportRowModelApi.ts
 - Update main.ts to export IViewportRowModel interface
 - Adjust imports in viewportRowModelApi.ts to remove server-side dependencies

sdwvit committed:
Introduce shared API for row height management across row models

- Added `resetRowHeights` and `onRowHeightChanged` to `iRowModel` interface
- Consolidated row model shared functions into new `rowModelSharedApi.ts` file
- Updated module dependencies to include `RowModelSharedApiModule`
- Implemented row height methods in client, server, and viewport row models
- Extended interfaces to inherit shared row model API functions

sdwvit committed:
Add viewport row model API and update onRowHeightChanged validation

 - Add _ViewportRowModelGridApi with onRowHeightChanged and resetRowHeights methods
 - Update apiFunctionValidator.ts to include 'viewport' in onRowHeightChanged targets
 - Add tsconfig$paths exclusion in jsLibraryMappings.xml

sdwvit committed:
Add row height state management with size limit in viewport row model

- Add HEIGHT_STATE_SIZE_LIMIT to cap variable row height entries
- Introduce heightsState Map for tracking non-default row heights
- Implement getRowTop to calculate dynamic row positions
- Update updateRowHeights to manage height state and cleanup
- Modify forEachNode to include index parameter for accurate processing

sdwvit committed:
Implement binary search for row index calculation and variable height handling

- Replace division with binary search for accurate row index in variable height rows
- Add getRowBounds to retrieve dynamic row heights from Map
- Update getRowTop to account for variable row heights via Map entries

sdwvit committed:
Add viewport row model support to autoHeight

 - Import IViewportRowModel for type consistency
 - Update onRowHeightChanged to include viewport row model
 - Add forEachDisplayedNode with index parameter for processing
 - Extend autoHeight validation to include viewport row model
 - Remove outdated server-side row height note from documentation

sdwvit committed:
**Enable dynamic height for viewport row model and update documentation notes**

 - Update row-models.json to enable dynamic row height for viewport row model
 - Remove outdated note about variable row height in viewport model

sdwvit committed:
Switch to IRowModel and remove dynamic height support

 - Update class to implement IRowModel instead of IViewportRowModel
 - Remove heightsState map and related dynamic height tracking logic
 - Simplify getRowBounds to return fixed row height calculation
 - Replace binary search with direct pixel/rowHeight division
 - Add comments explaining removal of dynamic height support
 - Update rowTop calculation to use fixed row height multiplier
 - Remove obsolete HEIGHT_STATE_SIZE_LIMIT constant
 - Simplify forEachNode to only require rowNode and index parameters

sdwvit committed:
Remove ViewportRowModelApi module and disable dynamic height support

 - Remove _ViewportRowModelGridApi import from gridApiFunctions.ts
 - Disable viewport dynamic height in row-models.json
 - Update iModule.ts to remove ViewportRowModelApi from module names
 - Remove ViewportRowModelApiModule export from enterprise main.ts
 - Update onRowHeightChanged to remove warning for viewport model
 - Disable viewport row height state management in documentation notes

sdwvit committed:
Remove viewport from autoHeight supported row models

 - Update autoHeight validation to exclude viewport row model
 - Add check for paginationAutoPageSize in autoHeight validation

sdwvit committed:
Remove outdated comment on dynamic row height support in viewport model

- Remove comment explaining lack of dynamic row height support

sdwvit committed:
Remove console logs and add resetting row heights example

- Remove console logs from datasource and fake server implementations
- Add new "resetting-row-height" example with dynamic height calculation and reset functionality
- Update documentation to explain row height management patterns
- Introduce getRowHeight with sinusoidal pattern and onRowClicked height adjustment
- Add reset button and row height state management in new example implementation

sdwvit committed:
new line

AndrewGlazier commented on documentation/ag-grid-docs/src/content/docs/server-side-model-row-height/index.mdoc:
```diff
@@ -34,6 +31,19 @@ details on these properties.
 
+## Setting and restoring Row Height
+
+To dynamically set or restore row heights in the Server-Side Row Model, use `setRowHeight()` to apply custom heights to specific rows and `resetRowHeights()` to revert all rows to the values calculated by the `getRowHeight()` in Grid Options. See [Changing Row Height](https://localhost:4610/javascript-data-grid/row-height/#rownodesetrowheightheight-and-apionrowheightchanged) for more.
```
localhost (links should be relative also)

sdwvit replied:
ty, well spotted

AndrewGlazier commented on documentation/ag-grid-docs/src/content/docs/server-side-model-row-height/_examples/auto-row-height/fakeServer.ts:
```diff
@@ -19,8 +19,6 @@ export function FakeServer(allData) {
     function executeQuery(request) {
         const sql = buildSql(request);
 
-        console.log('[FakeServer] - about to execute query:', sql);
```
We leave these in as they're helpful for people to learn what the link between the grid interface and SQL is

sdwvit replied:
The purpose of these examples is to show dynamic row height, and console boxes only clutter this doc page. Do you insist on reverting this?

sdwvit committed:
Update grid options for row height management

 - Remove filter and enable properties from columnDefs
 - Add sortable: false to column defaults
 - Introduce getRowId and getRowHeight with sinusoidal calculation

sdwvit committed:
- Fix link to changing row height guide

sdwvit committed:
revert console calls
